nexusutils/nexus/nexus.py

```diff
@@ -1104,32 +1126,109 @@ def axis_helper(dim, nxdata, signal, axes, logger):
+    """ Get concept or nxdl path from given hdf_node.
+    """
+    # The bellow logger is for deactivatine unnecessary degub message above
+    if logger is None:
+        logger = logging.getLogger(__name__)
+        logger.setLevel(logging.INFO)
+    (_, _, nxdl_path) = get_nxdl_doc(hdf_node, logger, None)
+    con_path = ''
+    if nxdl_path:
+        for nd_ in nxdl_path:
+            con_path = con_path + '/' + get_node_name(nd_)
+    return con_path
+
+
 class HandleNexus:
     """documentation"""
-    def __init__(self, logger, args):
+    def __init__(self, logger, nexus_file,
+                 d_inq_nd, c_inq_nd):
         self.logger = logger
         local_dir = os.path.abspath(os.path.dirname(__file__))
-        self.input_file_name = args[0] if len(
-            args) >= 1 else os.path.join(local_dir, '../../tests/data/nexus/201805_WSe2_arpes.nxs')
+
+        self.input_file_name = nexus_file if nexus_file is not None else \
+            os.path.join(local_dir, '../../tests/data/nexus/201805_WSe2_arpes.nxs')
         self.parser = None
         self.in_file = None
+        self.d_inq_nd = d_inq_nd
+        self.c_inq_nd = c_inq_nd
+        # Aggregating hdf path corresponds to concept query node
+        self.hdf_path_list_for_c_inq_nd = []
 
     def visit_node(self, hdf_name, hdf_node):
         """Function called by h5py that iterates on each node of hdf5file.
         It allows h5py visititems function to visit nodes."""
-        hdf_path = '/' + hdf_name
-        process_node(hdf_node, hdf_path, self.parser, self.logger)
+        if self.d_inq_nd is None and self.c_inq_nd is None:
+            process_node(hdf_node, '/' + hdf_name, self.parser, self.logger)
+        elif (self.d_inq_nd is not None
+              and (hdf_name == self.d_inq_nd)):
+            process_node(hdf_node, '/' + hdf_name, self.parser, self.logger)
+        elif self.c_inq_nd is not None:
+            ren_con = get_all_is_a_rel_from_hdf_node(hdf_node, None)
+            if self.c_inq_nd in ren_con:
+                self.hdf_path_list_for_c_inq_nd.append(hdf_name)
 
     def process_nexus_master_file(self, parser):
         """Process a nexus master file by processing all its nodes and their attributes"""
         self.parser = parser
         self.in_file = h5py.File(self.input_file_name, 'r')
         self.in_file.visititems(self.visit_node)
-        get_default_plotable(self.in_file, self.logger)
+        if self.d_inq_nd is None and self.c_inq_nd is None:
+            get_default_plotable(self.in_file, self.logger)
+        # To log the provided concept and concepts founded
+        if self.c_inq_nd is not None:
+            for hdf_path in self.hdf_path_list_for_c_inq_nd:
+                self.logger.info(hdf_path)
         self.in_file.close()
 
 
-def main():
+@click.command()
+@click.option(
+    '--nexus-file',
+    required=False,
+    default=None,
+    help=('NeXus file with extension .nxs to learn NeXus different concept'
```

Comment:
I think these help strings should also contain the slightly more verbose explanation from the README